concurrency/utils.c

```diff
@@ -1,43 +1,66 @@
 #include "utils.h"
+#include "semaphores.h"
+
+#define MAX_ERROR_MESSAGE_LENGTH 200
 
 // Reference to external function vcHalt (in vcuserlibrary.c)
 extern void vcHalt(int exitCode);
 
+//Used to manage sem value changes, and for error handler
+CSSem *vizconSem;
+
+//Used to create a semaphore that manages sem value changes, and errors
+void vizconSemCheck()
+{
+    if(vizconSem != NULL)
+        semClose(vizconSem);
+        vizconSem = NULL;
```

Comment:
What does this `if` need to cover? are there missing braces?

Reply:
There was missing braces, this is now fixed